src/context.rs

```diff
@@ -76,6 +77,120 @@ impl Context {
             0
         }
     }
+
+    /// Return the stock string for the [dc_stock::StringId].
+    ///
+    /// If the context callback responds with a string to use, e.g. a
+    /// translation, then this string will be returned.  Otherwise a
+    /// default (English) string is returned.
+    pub fn stock_str(&self, id: dc_stock::StringId) -> String {
```

Comment:
I think I would rather see these methods to be defined on `StringId` instead, that keeps concerns a bit more separated. Like this

```rust 
impl StringId {
	pub fn stock_str(&self, ctx: &Context) -> String {
			let ptr = ctx.call_cb(Event::GET_STRING, id as usize, 0) as *mut libc::c_char;
        if ptr.is_null() {
            self.default_string(id)
        } else {
            let ret = to_string(ptr);
            unsafe { free(ptr as *mut libc::c_void) };
            ret
        }
	}
}
```